### DISCUSSION TITLE
Copy a slice of audio playing in the browser and together with that the portion of its peaks data

### DISCUSSION THREAD

**Discussion by Manfredt** [katspaugh/wavesurfer.js#3338] [Q&A]
Thanks for this wonderful library which i recently have started using in my project. I am adding a feature to my audio player
where users would be able to copy a portion of the audio playing in the browser, let's say between the seconds range 12-35. To do this i am grabbing the url of the audio on the server, download the whole audio, and slice the audio using ffmpeg. This works nice. But i am struggling and now i have given up on how to create the audio peaks on the nodejs server. I know there are suggestions using audiowaveform but i failed to install this on google app engine. The ideal solution is that when the user copies a slice of the audio playing in the browser i would also like to copy ONLY the portion of the peaks data corresponding to that portion of the audio. If this is impossible is there an easier way to generate audio peaks data on a nodejs back end server? Thank you again for this wonderful library. I would be eternally grateful for any assistance provided. Your most humble servant.

![zemz_audio_player](https://github.com/katspaugh/wavesurfer.js/assets/7483544/054d653c-a4bd-47b3-9a4d-afc056155d29)

**Comment by katspaugh** [comment]
Something like [node-pcm](https://github.com/jhurliman/node-pcm) should work. It uses ffmpeg under the hood.

**Reply by Manfredt** [reply]
I have tried that and i could not make it work. Could you provide some hints?

**Reply by Manfredt** [reply]
Here is my attempt. I am stuck and there is scant documentation:

![image](https://github.com/katspaugh/wavesurfer.js/assets/7483544/b081d084-3a07-4192-a25f-2e9a63eb993a)
